### DISCUSSION TITLE
The document increases in size according to the accesses.

### DISCUSSION THREAD

**Discussion by echevarria** [vercel/next.js#14322] [Help]
[](url)
![image](https://user-images.githubusercontent.com/1697362/85030738-33a45b80-b154-11ea-80ee-d0f037f8c5c6.png)

## Problem

I deployed the application on server and run in production mode. When starts the document size is 16k but when users start to access it increases in size, in the figure it has 12MB. 

What's happening? I need do a different configuration on the application?

It's just happens on production mode.

**Comment by timneutkens** [comment]
Sounds like you're leaking data between requests which increases the size with every new request (sort of like a memory leak). Potentially you have a global variable in the server environment.

**Reply by maiconkf** [reply]
> Sounds like you're leaking data between requests which increases the size with every new request (sort of like a memory leak). Potentially you have a global variable in the server environment.

can you give an example, please?

this is the server.js:
```javascript
const express = require("express");
const next = require("next");
const port = parseInt(process.env.PORT, 10) || 3000;
const dev = process.env.NODE_ENV !== "production";
const app = next({ dev });
const handle = app.getRequestHandler();
const path = require("path");

app.prepare().then(() => {
  const server = express();
  server.get("/ads.txt", (req, res) => {
    const filePath = path.join(__dirname, "static/ads.txt");
    app.serveStatic(req, res, filePath);
  });

  server.get("*", (req, res) => handle(req, res));
  server.listen(port, (err) => {
    if (err) throw err;
    console.log(`🤘 on http://localhost:${port}`);
  });
});
```

**Reply by timneutkens** [reply]
## ⚠️ This code is effectively a memory leak just for demonstration purposes, do not add it to your application ⚠️ 

E.g. you can have code like:


```jsx
let myData = { items: [] }

export function getServerSideProps() {
  const item = await getDataFromSomewhere()

  myData.items.push(item)
  return {
    props: {
      myData
    }
  }
}

export default ({myData}) => {
  return <>
    <h1>Memory leak demo do not use in production</h1>
    <ul>
	{myData.map(item => <li>{item.id}</li>)}
    </ul>
  </>
}
```

Every request would add something into the module scope variable, this means that at request 1 your get 1 item, 2 you get 2 items and so on and so forth.

The best way to track it down in your application is doing a diff of the output before/after, you'll be able to see what is increasing the size of the document
